src/graph/graphcontext.c

```diff
@@ -660,6 +660,113 @@ int GraphContext_DeleteIndex
 	return res;
 }
 
+// remove a single node from all indices that refer to it
+void _DeleteNodeFromIndices
+(
+	GraphContext *gc,  // graph context
+	Node *n,           // node to remove from index
+	LabelID *lbls,     // [optional] node labels to remove from index
+	uint label_count   // [optional] number of labels
+) {
+	ASSERT(n  != NULL);
+	ASSERT(gc != NULL);
+	ASSERT(lbls != NULL);
+
+	Schema   *s      = NULL;
+	EntityID node_id = ENTITY_GET_ID(n);
+
+	for(uint i = 0; i < label_count; i++) {
+		int label_id = lbls[i];
+		ASSERT(Graph_IsNodeLabeled(gc->g, ENTITY_GET_ID(n), label_id));
+		s = GraphContext_GetSchemaByID(gc, label_id, SCHEMA_NODE);
+		ASSERT(s != NULL);
+
+		// update any indices this entity is represented in
+		Schema_RemoveNodeFromIndex(s, n);
+	}
+}
+
+// remove a single node from all indices that refer to it
+void GraphContext_DeleteNodeFromIndices
+(
+	GraphContext *gc,  // graph context
+	Node *n,           // node to remove from index
+	LabelID *lbls,     // [optional] node labels to remove from index
+	uint label_count   // [optional] number of labels
+) {
```

Comment:
`GraphContext_AddNodeToIndices` seems to be calling `NODE_GET_LABELS` why can't we do the same here?